lib/packwerk/offense_collection.rb

```diff
@@ -71,17 +71,28 @@ def outstanding_offenses
 
     private
 
-    sig { params(package: Packwerk::Package).returns(Packwerk::DeprecatedReferences) }
-    def deprecated_references_for(package)
-      @deprecated_references[package] ||= Packwerk::DeprecatedReferences.new(
+    sig { params(package: Packwerk::Package).returns(Packwerk::PackageTodo) }
+    def package_todo_for(package)
+      @package_todo[package] ||= Packwerk::PackageTodo.new(
         package,
-        deprecated_references_file_for(package),
+        package_todo_file_for(package),
       )
     end
 
     sig { params(package: Packwerk::Package).returns(String) }
-    def deprecated_references_file_for(package)
-      File.join(@root_path, package.name, "deprecated_references.yml")
+    def package_todo_file_for(package)
+      if Pathname.new(@root_path).join(package.name, "deprecated_references.yml").exist?
+        warning = <<~WARNING
+          DEPRECATION WARNING: `deprecated_references.yml` files have been renamed to
+          `package_todo.yml`. Please see https://github.com/Shopify/packwerk/releases
```

Comment:
- Is it possible to rename them automatically when we run check or update?
- You'll want a require for `Pathname` at the top of this file since we're using it now.
- Since we depend on Active Support, we should probably leverage it for deprecation warnings.

Reply:
> It is possible, but I'd prefer not to do this. I think this would be really surprising for users.

If you log a message when it happens explaining the change I don't feel like it would be confusing. We wouldn't have to code an extra command or point to release notes if it _just worked_.

> This is the same API we use in lib/packwerk/sanity_checker.rb and lib/packwerk/configuration.rb for warning about deprecated API, so I wanted to be consistent with that.

Noted. We should be consistent 👍

Reply:
Okay! I'm down to play around with it in a separate PR. We can feel out what makes most sense there.